editors/vscode/Portfile

```
@@ -0,0 +1,78 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+
+github.setup            microsoft vscode 1.63.2
+revision                0
+categories              editors
+maintainers             {@sainnhe gmail.com:sainnhe} \
+                        openmaintainer
+license                 MIT
+supported_archs         x86_64 arm64
+
+description             Visual Studio Code - Open Source (Code - OSS)
+
+long_description \
+    Visual Studio Code combines the simplicity of a code editor with what developers \
+    need for their core edit-build-debug cycle. It provides comprehensive code editing, \
+    navigation, and understanding support along with lightweight debugging, a rich \
+    extensibility model, and lightweight integration with existing tools.
+
+github.tarball_from     archive
+
+checksums               rmd160  179bf6d4b2738022d8fa7ccedf509ee83898bbf9 \
+                        sha256  21fc9bc17ba4cf480b1e006f298363d86215c339c480f8d781cabcfedad2d624 \
+                        size    15411738
+
+depends_build-append    port:nodejs16 \
+                        port:npm8 \
+                        port:yarn
+
+use_configure           no
+
+patchfiles              patch-postinstall-fix.diff
+
+variant stable conflicts insiders description {Patch with features in VSCode stable release} {
+    patchfiles-append   patch-features-stable.diff
+}
+
+variant insiders conflicts stable description {Patch with features in VSCode insiders release} {
+    patchfiles-append   patch-features-insiders.diff
+}
+
+default_variants        +stable
+universal_variant       no
+
+if {${build_arch} eq "arm64"} {
+    set vscode-arch "arm64"
+} else {
+    set vscode-arch "x64"
+}
+
+build {
+    if {[variant_isset stable]} {
+        system "/usr/bin/curl -sL https://github.com/dhanishgajjar/vscode-icons/raw/master/icns/default.icns -o ${worksrcpath}/resources/darwin/code.icns"
+    } elseif {[variant_isset insiders]} {
+        system "/usr/bin/curl -sL https://github.com/dhanishgajjar/vscode-icons/raw/master/icns/default_insider_light.icns -o ${worksrcpath}/resources/darwin/code.icns"
+    }
+    system "export PYTHON=/usr/bin/python3 && cd ${worksrcpath} && yarn && yarn gulp vscode-darwin-${vscode-arch}"
```

Don't `cd` within `system`; use the `-W` argument instead.
```suggestion
    system -W ${worksrcpath} "export PYTHON=/usr/bin/python3 && yarn && yarn gulp vscode-darwin-${vscode-arch}"
```